### PR TITLE
added solution for homework Palindrome

### PR DESCRIPTION
**Pull request by Valdes2**

### CODE REVIEW

File: src/main/java/core/basesyntax/Palindrome.java [hunk 1]
```diff
@@ -21,6 +21,11 @@
      * <p>Результат: true</p>
      */
     public boolean isPalindrome(String text) {
-        return false;
+        StringBuilder reverseText = new StringBuilder();
+        String textBuf = text;
+        textBuf = textBuf.replaceAll("[^a-zA-Z0-9]","");
+        text = text.replaceAll("[^a-zA-Z0-9]","");
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Palindrome.java:24]: @Valdes2 avoid using replaceAll twice

Comment (Valdes2) [reply src/main/java/core/basesyntax/Palindrome.java:24]: why is that? bad practice?

Comment (Valdes2) [reply src/main/java/core/basesyntax/Palindrome.java:24]: Ooo, i realized! Sorry, stupid question.

File: src/main/java/core/basesyntax/Palindrome.java [hunk 2]
```diff
@@ -21,6 +21,8 @@
      * <p>Результат: true</p>
      */
     public boolean isPalindrome(String text) {
-        return false;
+        StringBuilder reverseText = new StringBuilder(text = text.replaceAll("[^a-zA-Z0-9]",""));
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Palindrome.java]: @Valdes2 try to not initialize two variables in one statement
```suggestion
        text = text.replaceAll("[^a-zA-Z0-9]","")
        StringBuilder reversedText = new StringBuilder(text).reverse();
```

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Palindrome.java (hunk 2 above)
Comment (Valdes2) [reply src/main/java/core/basesyntax/Palindrome.java]: fixed

**Pull request closed without merging**